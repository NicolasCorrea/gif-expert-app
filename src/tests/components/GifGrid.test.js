import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");


describe('test <GifGrid />', () => {

    test('should render', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={"One Piece"} />);
        expect(wrapper).toMatchSnapshot();
    })


    test('should show item whe load images from useFechGifs', () => {
        const gifs = [
            {
                id: "ABC1",
                url: "http://locahost/image.png",
                title: "something"
            },
            {
                id: "ABC2",
                url: "http://locahost/image.png",
                title: "something2"
            }
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={"One Piece"} />);
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);


    })


})
