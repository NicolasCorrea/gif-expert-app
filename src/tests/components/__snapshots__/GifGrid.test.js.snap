// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <GifGrid /> should render 1`] = `
<Fragment>
  <h3>
    One Piece
  </h3>
  Loading...
  <div
    className="card-grid"
  />
</Fragment>
`;
